Framework/CurveFitting/src/FitMW.cpp

```diff
@@ -198,15 +198,26 @@ void FitMW::createDomain(boost::shared_ptr<API::FunctionDomain> &domain,
     // else continue with simple domain
   }
 
-  // set function domain
-  domain.reset(new API::FunctionDomain1DSpectrum(m_workspaceIndex, from, to));
+  if (m_matrixWorkspace->isHistogramData()) {
+    std::vector<double> x(static_cast<size_t>(to - from));
+    auto it = from;
+    for (size_t i = 0; it != to; ++it, ++i) {
+      x[i] = (*it + *(it + 1)) / 2;
+    }
+    domain.reset(new API::FunctionDomain1DSpectrum(m_workspaceIndex, x));
+    x.clear();
+  } else {
+    domain.reset(new API::FunctionDomain1DSpectrum(m_workspaceIndex, from, to));
+  }
```

Comment:
No I found that just replacing all references to the X,Y and E vectors with the HistogramX/Y/E types worked (without modifying the behaviour). Not sure why there is such a huge difference in the Fit results but I decided to make a note and move on to something else.